Add September monthly update

kagamiori opened the pull request:
Differential Revision: D49847351

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,93 @@
+Presto Functions
+================
+
+* Add bitwise_xor_agg aggregate function :pr:`6705`
```
- Let's group scalar, aggregate and window functions together. Let's also group casts together.
- Let's add links to function names using :func:

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,93 @@
+Build Systems
+=============
+
+* Install dependencies needed to support Azure Storage ABFS Connector :pr:`6418`
```
Can we list these dependencies explicitly? CC: @kgpai

kagamiori replied:
This PR installs azure-sdk-for-cpp and its dependencies openssl and perl-IPC-Cmd.

kgpai replied:
Thanks , azure/azure-sdk-for-cpp azure-storage-blobs_12.8.0 is the main dependency. We can say azure storage sdk.

kagamiori replied:
It also build sdk/core/azure-core though.

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,93 @@
+==============
+
+* Add native Parquet writer :pr:`6586`
+* Add S3FileSystem Write support :pr:`6021`
```
perhaps, Add support for writing query results to S3

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,93 @@
+
+* Add BaseVector::containsNullAt API :pr:`6515`
+* Add ContainerRowSerde::compareWithNulls API :pr:`6419`
+* Add default memory reclaimer for exchange client :pr:`6759`
```
Let's remove this. Too much of implementation details.

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,93 @@
+* Add support for Array/Map/RowVector::copyRanges from UNKNOWN source :pr:`6607`
+* Add support for HUGEINT in function signatures :pr:`6799`
+* Add support for prioritizing productive sources in Exchange :pr:`6459`
+* Add support for UNKNOWN types to VectorHasher::hash :pr:`6435`
```
Let's remove. Too much details.

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,93 @@
+* Add support for HUGEINT in function signatures :pr:`6799`
+* Add support for prioritizing productive sources in Exchange :pr:`6459`
+* Add support for UNKNOWN types to VectorHasher::hash :pr:`6435`
+* Add support for writing sorted buckted hive table :pr:`6142`
```
typo: buckted -> bucketed

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,93 @@
+* Add support for prioritizing productive sources in Exchange :pr:`6459`
+* Add support for UNKNOWN types to VectorHasher::hash :pr:`6435`
+* Add support for writing sorted buckted hive table :pr:`6142`
+* Extend Type with isOrderable() and isComparable() :pr:`6770`
```
nit: Add Type::isOrderable() and Type::isComparable() APIs

mbasmanova reviewed (commented):
@kagamiori Wei, thank you for putting together this update. Some comments below.

kagamiori commented:
> @kagamiori Wei, thank you for putting together this update. Some comments below.

Hi @mbasmanova, Thank you for reviewing. I've addressed your comments and updated the PR. Please take a look. Thanks!

kgpai reviewed (commented):
Thanks Wei, looks great.

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,98 @@
+==============
+
+* Add support for writing query results to S3 :pr:`6021`
+* Fix Parquet requestType to align with scan table schema :pr:`6404`
```
It is not clear what this means. Would it be possible to re-write in user-friendly terms?

kagamiori replied:
According to my understanding, this PR passes the expected output type of TableScan to Parquet column reader (https://github.com/facebookincubator/velox/issues/5770). I can double check with @Yuhta tomorrow.

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,98 @@
+===========================
+
+* Add fast path of FlatMap column reader for scalar types :pr:`6507`
+* Fix :func:`array_remove` null handling to follow Presto's behavior :pr:`6424`
```
Should this go into Presto Functions section?

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,98 @@
+* Add fast path of FlatMap column reader for scalar types :pr:`6507`
+* Fix :func:`array_remove` null handling to follow Presto's behavior :pr:`6424`
+* Fix CPU stats tracking by attributing LazyVector loading time to TableScan :pr:`6558`
+* Fix handling null string in :func:`json_extract` to follow Presto's behavior :pr:`6439`
```
Move to Presto Functions section?

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,98 @@
+* Fix :func:`array_remove` null handling to follow Presto's behavior :pr:`6424`
+* Fix CPU stats tracking by attributing LazyVector loading time to TableScan :pr:`6558`
+* Fix handling null string in :func:`json_extract` to follow Presto's behavior :pr:`6439`
+* Fix :func:`min`/:func:`max`/:func:`min_by`/:func:`max_by`` Presto aggregates when input contains nested nulls :pr:`6723`
```
+1 

Move to Presto Functions section?

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,98 @@
+* Fix handling null string in :func:`json_extract` to follow Presto's behavior :pr:`6439`
+* Fix :func:`min`/:func:`max`/:func:`min_by`/:func:`max_by`` Presto aggregates when input contains nested nulls :pr:`6723`
+* Fix :func:`set_union` Presto aggregate to follow Presto's behavior :pr:`6800`
+* Fix tracking of the number of processed rows in Window operator :pr:`6584`
```
Remove. Too much detail.

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,98 @@
+* Fix :func:`url_extract_path` Presto function :pr:`6657`
+* Optimize exception handling in SIMDJsonExtractor parse and extract :pr:`6776`
+* Optimize large chunk memory allocation in stream arena :pr:`6547`
+* Optimize :func:`reduce_agg` aggregation :pr:`6536`
```
Remove.

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,98 @@
+* Optimize :func:`reduce_agg` aggregation :pr:`6536`
+* Optimize spill performance by buffering spill data write :pr:`6509`
+* Optimize TRY_CAST performance by reducing exception throws :pr:`5913`
+* Reduce peak memory usage by roughly half in HiveDataSource :pr:`6601`
```
Move to Hive Connector section

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,96 @@
+* Add support for Array/Map/RowVector::copyRanges from UNKNOWN source :pr:`6607`
+* Add support for HUGEINT in function signatures :pr:`6799`
+* Add support for prioritizing productive sources in Exchange :pr:`6459`
+* Add support for writing sorted bucketed hive table :pr:`6142`
```
Can you move this to Hive Connector section?

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,96 @@
+* Add :func:`reduce_agg` aggregate function :pr:`6482`
+* Add support for UNKNOWN type in :func:`arbitrary` aggregate function :pr:`6557`
+* Add support for complex types to compare argument of :func:`min_by` and :func:`max_by` :pr:`6605`
+* Fix :func:`array_remove` null handling to follow Presto's behavior :pr:`6424`
```
"to follow Presto's behavior" is redundant. Please, remove here and in other places.

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,96 @@
+Performance and Correctness
+===========================
+
+* Add fast path of FlatMap column reader for scalar types :pr:`6507`
```
Move this to Hive Connector

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,96 @@
+===========================
+
+* Add fast path of FlatMap column reader for scalar types :pr:`6507`
+* Fix CPU stats tracking by attributing LazyVector loading time to TableScan :pr:`6558`
```
"CPU stats tracking" -> "CPU time reporting"

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,96 @@
+
+* Add fast path of FlatMap column reader for scalar types :pr:`6507`
+* Fix CPU stats tracking by attributing LazyVector loading time to TableScan :pr:`6558`
+* Optimize exception handling in SIMDJsonExtractor parse and extract :pr:`6776`
```
Perhaps, replace " SIMDJsonExtractor parse and extract " with the names of Presto functions that use this code path. Consider moving to Presto Functions section.

kagamiori replied:
All SIMD JSON functions would be affected and all our JSON functions except json_format and json_parse are SIMD JSON functions. I updated it to "Optimize exception handling in SIMD JSON functions".

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,96 @@
+* Optimize exception handling in SIMDJsonExtractor parse and extract :pr:`6776`
+* Optimize large chunk memory allocation in stream arena :pr:`6547`
+* Optimize spill performance by buffering spill data write :pr:`6509`
+* Optimize TRY_CAST performance by reducing exception throws :pr:`5913`
```
typos in "reducing exception throws "

mbasmanova commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,96 @@
+* Optimize large chunk memory allocation in stream arena :pr:`6547`
+* Optimize spill performance by buffering spill data write :pr:`6509`
+* Optimize TRY_CAST performance by reducing exception throws :pr:`5913`
+* Tune memory arbitration and spilling for Velox batch use cases :pr:`6789`
```
Group all spilling updates together.

mbasmanova reviewed (commented):
Looks great % a few remaining nits.

mbasmanova reviewed (approved):
Thank you, Wei.

aditi-pandit commented on velox/docs/monthly-updates/september-2023.rst:
```diff
@@ -0,0 +1,96 @@
+* Fix :func:`url_extract_path` function :pr:`6657`
+* Fix :func:`min`, :func:`max`, :func:`min_by`, and :func:`max_by` aggregates when input contains nested nulls :pr:`6723`
+* Fix :func:`set_union` aggregate :pr:`6800`
+* Fix :func:`lead` and :func:`lag` window function for int64 offset :pr:`6463`
```
This update isn't needed really.

aditi-pandit reviewed (approved):
Thanks @kagamiori for the updates !

xiaoxmeng reviewed (commented):
@kagamiori LGTM. Thanks!
